addons/de/xrechnung/instructions.go

```diff
@@ -0,0 +1,107 @@
+package xrechnung
+
+import (
+	"github.com/invopop/gobl/bill"
+	"github.com/invopop/gobl/cbc"
+	"github.com/invopop/gobl/pay"
+	"github.com/invopop/validation"
+)
+
+// Payment keys for XRechnung SEPA direct debit and credit transfer
+const (
+	KeyPaymentMeansSEPACreditTransfer cbc.Key = "sepa-credit-transfer"
+	KeyPaymentMeansSEPADirectDebit    cbc.Key = "sepa-direct-debit"
+)
+
+var validPaymentKeys = []cbc.Key{
+	pay.MeansKeyCash,
+	pay.MeansKeyCheque,
+	pay.MeansKeyCreditTransfer,
+	pay.MeansKeyCard,
+	pay.MeansKeyDirectDebit,
+	pay.MeansKeyOther,
+	KeyPaymentMeansSEPACreditTransfer,
+	KeyPaymentMeansSEPADirectDebit,
+}
+
+// ValidatePaymentInstructions validates the payment instructions according to the XRechnung standard
+func ValidatePaymentInstructions(value interface{}) error {
+	inv, ok := value.(*bill.Invoice)
+	if !ok || inv == nil || inv.Payment == nil || inv.Payment.Instructions == nil {
+		return nil
+	}
+	instr := inv.Payment.Instructions
+	return validation.ValidateStruct(instr,
+		validation.Field(&instr.Key,
+			validation.Required,
+			validation.By(validatePaymentKey),
+		),
+		// BR-DE-23
+		validation.Field(&instr.CreditTransfer,
+			validation.When(instr.Key == KeyPaymentMeansSEPACreditTransfer,
+				validation.Required,
+				validation.Each(validation.By(validateCreditTransfer)),
+			),
+		),
+		// BR-DE-24
+		validation.Field(&instr.Card,
+			validation.When(instr.Key == pay.MeansKeyCard,
+				validation.Required,
+			),
+		),
+		// BR-DE-25
+		validation.Field(&instr.DirectDebit,
+			validation.When(instr.Key == KeyPaymentMeansSEPADirectDebit || instr.Key == pay.MeansKeyDirectDebit,
+				validation.Required,
+				validation.By(validateDirectDebit),
+			),
+		),
+	)
+}
+
+func validatePaymentKey(value interface{}) error {
+	t, ok := value.(cbc.Key)
+	if !ok {
+		return validation.NewError("type", "Invalid payment key")
```

Comment:
I'm not sure about this usage of `NewError`, what is "`type`"? Also, convention is that error messages should not start with capital letters.